### DISCUSSION TITLE
How can i enter a subpopup?

### DISCUSSION THREAD

**Discussion by hrqmonteiro** [NeogitOrg/neogit#1494] [General]
I need to enter the "branch/revision" subpopup inside the Branch popup.

This one:

 
![image](https://github.com/user-attachments/assets/a5ce8029-09b3-4abe-bf05-2e7124521bdc)

I know i can do `:Neogit branch` to enter the Branch popup, but i want to do a keybinding that already enters the one to checkout into another branch

How can i do it?

**Comment by CKolkey** [comment]
Like so: https://github.com/NeogitOrg/neogit/blob/master/doc/neogit.txt#L583

**Reply by hrqmonteiro** [reply]
> Like so: https://github.com/NeogitOrg/neogit/blob/master/doc/neogit.txt#L583

I am trying the one from the example and nothing happens...


https://github.com/user-attachments/assets/5aa039b3-632b-4282-ac93-472d90f51a84

**Reply by CKolkey** [reply]
Ahh, the reason that did nothing is that it's returning a function - add `()` to the end of that like:

`:lua require("neogit").action("commit", "commit", { "--verbose", "--all" })()`

**Comment by hrqmonteiro** [comment]
Tried setting up directly in a keybinding

`{"<Leader>gcc", "neogit.action('branch', 'create_branch')", desc = "Commit"}`

But when i do it, it returns:

`E486: Pattern not found: trouble`

![image](https://github.com/user-attachments/assets/2550aa4e-a843-4b36-8bf0-c80d325349eb)

**Reply by CKolkey** [reply]
You wouldn't want that as a string, but as lua:

```lua
{"<Leader>gcc", neogit.action('branch', 'create_branch'), desc = "Commit"}
```

**Reply by rbhanot4739** [reply]
@CKolkey I get following errors when I run `:lua require("neogit").action("diff", "staged")()`

```
   Error  02:53:21 PM msg_show.lua_error   require("neogit").action("diff", "staged")() E5108: Error executing lua ...share/nvim/lazy/plenary.nvim/lua/plenary/async/async.lua:18: The coroutine failed with this message: ...hare/nvim/lazy/neogit/lua/neogit/popups/diff/actions.lua:63: attempt to call method 'close' (a nil value)
stack traceback:
	[C]: in function 'error'
	...share/nvim/lazy/plenary.nvim/lua/plenary/async/async.lua:18: in function 'callback_or_next'
	...share/nvim/lazy/plenary.nvim/lua/plenary/async/async.lua:45: in function 'step'
	...share/nvim/lazy/plenary.nvim/lua/plenary/async/async.lua:48: in function 'execute'
	...share/nvim/lazy/plenary.nvim/lua/plenary/async/async.lua:118: in function <...share/nvim/lazy/plenary.nvim/lua/plenary/async/async.lua:117>
	/home/rbhanot/.local/share/nvim/lazy/neogit/lua/neogit.lua:227: in function </home/rbhanot/.local/share/nvim/lazy/neogit/lua/neogit.lua:199>
	[string ":lua"]:1: in main chunk
```

The same error occurs for other popups and actions as well, for example the `commit` popup with `commit` action gave the same error.

It's super painful to having to press bunch of keys through popups to open a desired view.
